wasmsdk/blobber.go

```diff
@@ -839,11 +839,19 @@ func upload(allocationID, remotePath string, fileBytes, thumbnailBytes []byte, w
 }
 
 // download download file blocks
-func downloadBlocks(alloc *sdk.Allocation, remotePath, authTicket, lookupHash string, startBlock, endBlock int64) ([]byte, error) {
+func downloadBlocks(allocId string, remotePath, authTicket, lookupHash string, startBlock, endBlock int64) ([]byte, error) {
+
 	if len(remotePath) == 0 && len(authTicket) == 0 {
 		return nil, RequiredArg("remotePath/authTicket")
 	}
 
+	alloc, err := getAllocation(allocId)
```

Comment:
why do we need to re-request the allocation if we have an object already`?

Reply:
The wasm logged out `downloadBlocks binder not implemented` because `downloadBlocks` wasm function did’t work as it took `*sdk.Allocation` as parameter

so I passed the `allocId` string instead of `*sdk.Allocation` as parameter

Reply:
We cannot pass native go objects from js, so we need to pass allocationID, we have a LRU for allocations so no call will be made to sharders if alloction is already requested before.